src/main/java/org/jabref/model/entry/BibEntry.java

```diff
@@ -1044,4 +1047,42 @@ public void replaceDownloadedFile(String linkToDownloadedFile, LinkedFile downlo
 
         this.setFiles(linkedFiles);
     }
+
+    /**
+     * Merge this entry's fields with another BibEntry. Non-intersecting fields will be automatically merged. In cases of
+     * intersection, priority is given to THIS entry's field value.
+     *
+     * @param other another BibEntry from which fields are sourced from
+     */
+    public void mergeWith(BibEntry other) {
+        mergeWith(other, Set.of());
+    }
+
+    /**
+     * Merge this entry's fields with another BibEntry. Non-intersecting fields will be automatically merged. In cases of
+     * intersection, priority is given to THIS entry's field value, UNLESS specified otherwise in the arguments.
+     *
+     * @param other another BibEntry from which fields are sourced from
+     * @param otherPrioritizedFields collection of Fields in which 'other' has a priority into final result
+     */
+    public void mergeWith(BibEntry other, Set<Field> otherPrioritizedFields) {
+        Set<Field> thisFields = new TreeSet<>(Comparator.comparing(Field::getName));
+        Set<Field> otherFields = new TreeSet<>(Comparator.comparing(Field::getName));
+
+        thisFields.addAll(this.getFields());
+        otherFields.addAll(other.getFields());
+
+        Set<String> thisFieldsNames = thisFields.stream().map(Field::getName).collect(Collectors.toSet());
+        Set<String> otherPrioritizedFieldsNames = otherPrioritizedFields.stream().map(Field::getName).collect(Collectors.toSet());
+
+        for (Field otherField : otherFields) {
+            // As iterator only goes through non-null fields from OTHER, the value below can never be null
+            String otherFieldValue = other.getField(otherField).orElse(null);
+            if (!thisFieldsNames.contains(otherField.getName())) {
+                this.setField(otherField, otherFieldValue);
+            } else if (otherPrioritizedFieldsNames.contains(otherField.getName())) {
```

Comment:
A `contains` does not work with `Field` as data type? I wonder why the conversion from `Set<Field>` to `Set<String>` is necessary, when here also a `Field` is available.

Reply:
I made comparisons with Fields names (Strings) because, although all classes / enums that currently implement this interface spawn objects than _can_ be directly compared (in the `contains` method), the interface itself does not provide a guarantee that this would be the case (i.e. no `equals` (or similar) on `Field` interface).

If enums implement it, this would be no problem. The `UnknownField` class also implements an `equals()`. But future implementations might not. So I decided that using their name would be more reliable. A modification to `Field` itself could be an alternative, I suppose...

Reply:
(Also, simplified these _if / else_ cases to a single _if_ with both conditions separated by an OR operation)

Reply:
Adding `equals` to the interface is future work ^^

Reply:
I do think directly comparing `Field`s would result in a cleaner code, but for the sake of reliability, I'll be leaving like that (also, putting a short version of this reasoning as a comment).